src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerExtensions.cs

```diff
@@ -2,16 +2,131 @@
+                services.AddSingleton<ITelemetryModule, QuickPulseTelemetryModule>();
+
+                ServerTelemetryChannel serverChannel = new ServerTelemetryChannel();
+                services.AddSingleton<ITelemetryModule>(serverChannel);
+                services.AddSingleton<ITelemetryChannel>(serverChannel);
+                services.AddSingleton<TelemetryConfiguration>(provider =>
+                {
+                    ITelemetryChannel channel = provider.GetService<ITelemetryChannel>();
+                    TelemetryConfiguration config = new TelemetryConfiguration(instrumentationKey, channel);
+                    
+                    foreach (ITelemetryInitializer initializer in provider.GetServices<ITelemetryInitializer>())
+                    {
+                        config.TelemetryInitializers.Add(initializer);
+                    }
+
+                    QuickPulseTelemetryModule quickPulseModule = null;
+                    foreach (ITelemetryModule module in provider.GetServices<ITelemetryModule>())
+                    {
+                        if (module is QuickPulseTelemetryModule telemetryModule)
+                        {
+                            quickPulseModule = telemetryModule;
+                        }
+                        module.Initialize(config);
+                    }
+
+                    QuickPulseTelemetryProcessor processor = null;
+                    config.TelemetryProcessorChainBuilder
+                        .Use((next) =>
+                        {
+                            processor = new QuickPulseTelemetryProcessor(next);
+                            return processor;
+                        })
+                        .Use((next) => new FilteringTelemetryProcessor(filter, next));
+
+                    SamplingPercentageEstimatorSettings samplingSettings =
+                        provider.GetService<IOptions<SamplingPercentageEstimatorSettings>>()?.Value;
+
+                    if (samplingSettings != null)
+                    {
+                        config.TelemetryProcessorChainBuilder.Use((next) =>
+                            new AdaptiveSamplingTelemetryProcessor(samplingSettings, null, next));
+                    }
+
+                    config.TelemetryProcessorChainBuilder.Build();
+                    quickPulseModule?.RegisterTelemetryProcessor(processor);
+
+                    return config;
+                });
+                services.AddSingleton<TelemetryClient>(provider =>
+                {
+                    TelemetryConfiguration configuration = provider.GetService<TelemetryConfiguration>();
+                    TelemetryClient client = new TelemetryClient(configuration);
+
+                    string assemblyVersion = GetAssemblyFileVersion(typeof(JobHost).Assembly);
+                    client.Context.GetInternalContext().SdkVersion = $"webjobs: {assemblyVersion}";
```

Comment:
Cool. We'll need to do this in the Functions repo (we change this to "azurefunctions: " for better tracking) -- but this should be pretty easy.